python-lib/youtube_client.py

```diff
@@ -340,12 +342,21 @@ def store_next_page(self, url, headers, params, json_response):
         else:
             self.next_page = {}
 
+    def has_data_to_process(self):
+        if self.initial_data_process_loop:
+            self.initial_data_process_loop = False
+            return True
+        else:
+            return self.has_next_page()
+
     def has_next_page(self):
         return self.next_page is None or self.next_page != {}
```

Comment:
**!!!**
Should it not be 
`return self.next_page is not None and self.next_page != {}` ?